### PR TITLE
test: add create and drop sink test for nexmark

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously the `drop sink` can be stuck (not sure if it is stuck forever or comes back to normal after a long period of time)
After #8672, the `drop sink` is super fast.
We add `drop sink` into e2e tests to avoid similar problems from happening again in the future

## Checklist For Contributors

- ~[ ] I have written necessary rustdoc comments~
- [x] I have added necessary unit tests and integration tests
- ~[ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~
- ~[ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future.~ (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
There seem too many logs. Can we disable these?

```
2023-03-21T09:19:25.490674Z  INFO risingwave_frontend::scheduler::hummock_snapshot_manager: Unpin epoch 4073400070111232 with RPC

2023-03-21T09:14:24.546932Z  INFO risingwave_meta::hummock::manager: report_scale_compactor_info ScaleCompactorInfo { running_cores: 1, total_cores: 16, waiting_compaction_bytes: 0 } suggest_scale_out_core 0
```

**Comment by lmatz** [comment]
```
2023-03-21T09:20:27.670485Z  WARN risingwave_meta::rpc::election_client: error grpc request error: status: Unknown, message: "error reading a body from connection: stream closed because of a broken pipe", details: [], metadata: MetadataMap { headers: {} } received from leader observe stream
Tue Mar 21 09:20:27 UTC 2023 [risedev]: Program exited with 0
```

**Comment by BugenZhao** [comment]
image

File: ci/scripts/e2e-source-test.sh [hunk 1]
```diff
@@ -110,6 +111,7 @@ mysql --host=mysql --port=3306 -u root -p123456 < ./e2e_test/source/cdc/mysql_cd
 psql -h db -U postgres -d cdc_test < ./e2e_test/source/cdc/postgres_cdc_insert.sql
 
 # start cluster w/o clean-data
+RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info"
```
Comment (BugenZhao) [inline ci/scripts/e2e-source-test.sh:114]:
```suggestion
RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info" \
```

File: ci/scripts/e2e-source-test.sh [hunk 2]
```diff
@@ -122,13 +124,16 @@ pkill -f connector-node
 cargo make ci-kill
 
 echo "--- e2e, ci-1cn-1fe, nexmark endless"
+RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info"
```
Comment (BugenZhao) [inline ci/scripts/e2e-source-test.sh:127]:
```suggestion
RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info" \
```

File: ci/scripts/e2e-source-test.sh [hunk 3]
```diff
@@ -122,13 +124,16 @@ pkill -f connector-node
...
 cargo make ci-start ci-1cn-1fe
-sqllogictest -p 4566 -d dev './e2e_test/source/nexmark_endless/*.slt'
+sqllogictest -p 4566 -d dev './e2e_test/source/nexmark_endless_mvs/*.slt'
+sqllogictest -p 4566 -d dev './e2e_test/source/nexmark_endless_sinks/*.slt'
 
 echo "--- Kill cluster"
 cargo make ci-kill
 
 echo "--- e2e, ci-kafka-plus-pubsub, kafka and pubsub source"
+RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info"
```
Comment (BugenZhao) [inline ci/scripts/e2e-source-test.sh:136]:
```suggestion
RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info" \
```

File: ci/scripts/e2e-source-test.sh [hunk 4]
```diff
@@ -71,6 +71,7 @@ node_port=50051
 node_timeout=10
 
 echo "--- starting risingwave cluster with connector node"
+RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info"
```
Comment (BugenZhao) [inline ci/scripts/e2e-source-test.sh:74]:
```suggestion
RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info" \
```

**Comment by BugenZhao** [comment]
The `e2e-source` lasts too long and becomes the bottleneck of the workflow. 🥵 Is it possible to split it into another step?

File: ci/scripts/e2e-source-test.sh
Comment (xxchan) [inline ci/scripts/e2e-source-test.sh:1]: Is the speed expected?

image

> The e2e-source lasts too long and becomes the bottleneck of the workflow.

Can we move to `e2e-sink`?

**Comment by xxchan** [comment]
Wait, can we run the tests in parallel?! 🥵

File: ci/scripts/e2e-source-test.sh
Comment (tabVersion) [reply ci/scripts/e2e-source-test.sh:1]: +1

**Review by tabVersion — APPROVED** [review]
basically LGTM
lets keep watching the CI

**Comment by tabVersion** [comment]
> Wait, can we run the tests in parallel?! 🥵

Yes, any step behind `build` can run in parallel

**Comment by xxchan** [comment]
> > Wait, can we run the tests in parallel?! 🥵
> 
> Yes, any step behind `build` can run in parallel

I mean `sqllogictest -j`

File: ci/scripts/e2e-source-test.sh
Comment (lmatz) [reply ci/scripts/e2e-source-test.sh:1]: this is weird, I tested locally on my Mac, and it finishes very fast besides the 20-second sleep on purpose...
🤔 
After Bugen's two PR that makes `drop sink` completely async, `drop sink` is expected to be very fast.

This PR intends to detect this type of anomaly though.

**Comment by lmatz** [comment]
Succeed once: 
https://buildkite.com/risingwavelabs/pull-request/builds/21269#01876fa5-474c-41a5-8a35-ebcaa51afaa8

**Pull request merged**